src/kernel/mod.rs

```diff
@@ -47,15 +45,9 @@ pub trait Storage: Send + Sync + 'static + Sized {
     async fn remove(&self, key: &[u8]) -> Result<()>;
 
     /// 并行批量执行
-    #[inline]
-    async fn join(&self, vec_cmd: Vec<CommandData>) -> Result<Vec<Option<Vec<u8>>>> {
-        let map_cmd = vec_cmd.into_iter().map(|cmd| cmd.apply(self));
-        Ok(future::try_join_all(map_cmd)
-            .await?
-            .into_iter()
-            .map(CommandOption::into)
-            .collect_vec())
-    }
+    /// TODO need refactor
+    // #[inline]
+    // async fn join(&self, vec_cmd: Vec<CommandData>) -> Result<Vec<Option<Vec<u8>>>> {}
```

Comment:
mark

Reply:
Why not implement this function

Reply:
The return type of this function is not reasonable. For batch operations, we should tell client which part succeed and which part failed.

Reply:
sure

Reply:
> The return type of this function is not reasonable. For batch operations, we should tell client which part succeed and which part failed.

refer to the use of futures::try_join method, either all succeed at once, or all fail.